src/main/kotlin/sliv/tool/importer/view/ChoosingDirectoryView.kt

```diff
@@ -1,25 +1,43 @@
 package sliv.tool.importer.view
 
+import javafx.beans.property.SimpleStringProperty
+import javafx.collections.FXCollections
 import javafx.geometry.Insets
 import javafx.stage.DirectoryChooser
 import sliv.tool.importer.controller.ImporterController
 import tornadofx.*
+import java.io.File
+
 
 class ChoosingDirectoryView : Fragment() {
     private val controller: ImporterController by inject()
+    private val path = SimpleStringProperty()
+    private val files = FXCollections.observableArrayList<String>()
     private val directoryChooser = DirectoryChooser().apply { title = "Choose working directory" }
-    override val root = hbox(20) {
-        padding = Insets(5.0, 0.0, 0.0, 200.0)
-        label {
-            bind(controller.directoryPath)
+
+    override val root = vbox(20) {
+        padding = Insets(5.0, 10.0, 0.0, 10.0)
+        path.bindBidirectional(controller.directoryPath)
+        hbox(20){
+            label {
+                bind(controller.directoryPath)
+            }
+            button("Change") {
+                action {
+                    val dir = directoryChooser.showDialog(currentStage)
+                    controller.directoryPath.set(dir.absolutePath)
+                }
+            }
         }
-        button("Change") {
-            action {
-                val dir = directoryChooser.showDialog(currentStage)
-                controller.directoryPath.value = dir.absolutePath
+        path.onChange { op ->
+            for (file in File(op).walk()) {
+                if (!file.absolutePath.contains("/Images") && !file.isDirectory) {
```

Comment:
what is it? why not separate validation function? why images from capital letter? what if i will have path `C://images/nastya_molodets/dimaProject/images/1.jpg`?

Reply:
I thought that the names of folders in the project should be named in a certain way and in no other way

Reply:
My idea was that path to project can also contain such folder) it is not prohibited